### PR TITLE
Catch autoplay improvements

### PR DESCRIPTION
**Pull request by holly-hacker**
- Fixes crash with 2B hitobjects, which caused a -Infinity Catcher x position
- Use normal instead of interpolated movement for bananas when possible
- Ensure speedRequired is not NaN to prevent bugs in the future.

### CODE REVIEW

File: osu.Game.Rulesets.Catch/Replays/CatchAutoGenerator.cs [hunk 1]
```diff
@@ -44,9 +44,10 @@ void moveToNext(CatchHitObject h)
                 double timeAvailable = h.StartTime - lastTime;
 
                 //So we can either make it there without a dash or not.
-                double speedRequired = positionChange / timeAvailable;
+                double speedRequired = positionChange == 0 ? 0 : positionChange / timeAvailable;
```
Comment (iiSaLMaN) [inline osu.Game.Rulesets.Catch/Replays/CatchAutoGenerator.cs]: unnecessary, since 0 / timeAvailable = 0

**Review by frenzibyte — CHANGES_REQUESTED** [review]
small change

File: osu.Game.Rulesets.Catch/Replays/CatchAutoGenerator.cs (hunk 1 above)
Comment (HoLLy-HaCKeR) [reply osu.Game.Rulesets.Catch/Replays/CatchAutoGenerator.cs]: 0/0=NaN

Comment (peppy) [reply osu.Game.Rulesets.Catch/Replays/CatchAutoGenerator.cs]: The check should be on the second argument (denominator)

Comment (HoLLy-HaCKeR) [reply osu.Game.Rulesets.Catch/Replays/CatchAutoGenerator.cs]: Why exactly? If positionChange is non-zero and timeAvailable is zero, then speedRequired would be Infinity, as we want. If positionChange is zero, we never need to move so speedRequired should be zero.

Comment (peppy) [reply osu.Game.Rulesets.Catch/Replays/CatchAutoGenerator.cs]: please document that this is expected directly above this line, as this is not apparent (and the changed code looks like a coding error).

Comment (peppy) [reply osu.Game.Rulesets.Catch/Replays/CatchAutoGenerator.cs]: Updated the comment to specifically mention this. @smoogipoo worth double-checking this as to whether it is ambiguous in its current state.

Comment (smoogipoo) [reply osu.Game.Rulesets.Catch/Replays/CatchAutoGenerator.cs]: seems ok to me

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**